File: Maximum Number of Vowels in a Substring of Given Length.cpp
class Solution {
public:
    int maxVowels(string s, int k) {
        int maximum=0;
        deque<int>q;
        for(int i=0;i<k;i++){
            if(s[i]=='a' || s[i]=='e' ||s[i]=='i' || s[i]=='o' || s[i]=='u'){
                q.push_back(i);
            }
             
        }
        int num=q.size();
        maximum=max(num,maximum);
        for(int i=k;i<s.size();i++){
            if(!q.empty() && i-q.front()>=k){
                q.pop_front();
            }
            if(s[i]=='a' || s[i]=='e' ||s[i]=='i' || s[i]=='o' || s[i]=='u'){
                q.push_back(i);
            }
        
              int num=q.size();
             maximum=max(num,maximum);
            
            
        }
        return maximum;
    }
};

File: week-2/python/maximum-number-of-vowels-in-a-substring-of-given-length.py
class Solution:
    def maxVowels(self, s: str, k: int) -> int:
        vowels = {'a', 'e', 'i', 'o', 'u'}
        maximum = 0
        current_count = 0

        for i in range(k):
            if s[i] in vowels:
                current_count += 1

        maximum = max(current_count, maximum)

        for i in range(k, len(s)):
            if s[i - k] in vowels:
                current_count -= 1
            if s[i] in vowels:
                current_count += 1

            maximum = max(current_count, maximum)

        return maximum